crates/e2e/src/setup/onchain_components.rs

```diff
@@ -333,9 +335,19 @@ impl OnchainComponents {
                 .expect("getting accounts failed")[0],
             None,
         );
-        let tokens = self.deploy_tokens::<N>(minter).await;
+        let tokens = self.deploy_tokens::<N>(&minter).await;
+        self.seed_weth_uni_v2_pools(tokens.iter(), token_amount, weth_amount)
+            .await;
+        tokens
+    }
 
-        for MintableToken { contract, minter } in &tokens {
+    pub async fn seed_weth_uni_v2_pools<'a, I: Iterator<Item = &'a MintableToken>>(
```

Comment:
nit: Usually it's nicer to have the `IntoIterator` trait bound instead of `Iterator`. That allows you to pass iterators, slices and collections.
Instead of `token.iter()` you could then pass `&tokens` to `seed_with_uni_v2_pools()`.